cs/commandstation/UpdateProcessor.cxx

```diff
@@ -78,26 +78,31 @@ DECLARE_CONST(dcc_packet_min_refresh_delay_ms);
 
 StateFlowBase::Action UpdateProcessor::entry() {
   // We have an empty packet to fill. It is accessible in message()->data().
-  // First we check if there is an urgent update.
+  dcc::PacketSource* s = nullptr;
   Buffer<PriorityUpdate>* b = nullptr;
   {
     AtomicHolder h(this);
-    b = static_cast<Buffer<PriorityUpdate>*>(priorityUpdates_.next().item);
+    // First we check if there is an exclusive update.
+    if (has_exclusive()) {
+      s = refreshSources_[exclusiveIndex_];
+    } else {
+      // Then we check if there is an urgent update.
+      b = static_cast<Buffer<PriorityUpdate>*>(priorityUpdates_.next().item);
+    }
   }
   long long now = os_get_time_monotonic();
-  dcc::PacketSource* s = nullptr;
   unsigned code = 0;
   if (b) {
     // found a priority entry.
     s = b->data()->source;
     code = b->data()->code;
-    auto it = lastPacketTime_.find(s);
-    if (it == lastPacketTime_.end()) {
+    auto it = packetSourceStates_.find(s);
+    if (it == packetSourceStates_.end()) {
       // This packet source has been removed. Do not call it!
       b->unref();
       s = nullptr;
-    } else if (it->second >
-        now - MSEC_TO_NSEC(config_dcc_packet_min_refresh_delay_ms())) {
+    } else if (it->second.lastPacketTime_ >
+               now - MSEC_TO_NSEC(config_dcc_packet_min_refresh_delay_ms())) {
```

Comment:
Are you missing an extra parentheses on "now - MSEC_TO_NSEC(config_dcc_packet_min_refresh_delay_ms())"?  I don't now what the C/C++ syntax rules are here, but I always add the extra parentheses.

Reply:
done